README.md

```diff
@@ -216,11 +216,15 @@ The configuration file format depends on the version of jenkins-core and install
 Documentation is generated from a live instance, as well as a JSON schema you can use to validate configuration file
 with your favourite YAML tools.
 
+## Escape character
+If your need to put in an YAML definition a character which has a dedicated meaning (like character "$" or "'"), you need to escape it by character "^".
```

Comment:
I don't think this bit about yaml definition is correct, it is just in relation to secrets as far as i know...

Reply:
No, we use this approach with success in other areas like:

> tool:
>   jdk:
>     installations:
>       - name: "java1.9"
>         home: "^${NATIVE_TOOLS}^${SEP}^${JAVA_9}"

Reply:
thats because it thinks its a secret...

Reply:
You are right, I've take a look into the code.

On the other hand is there any limit/implication when we use the same behavior elsewhere? I couldn't find anything, but can overslooked sth. I think this escaping is useful not only for handling a secret.

Reply:
@pjanouse you should be able to write `home: "$NATIVE_TOOLS$SEP$JAVA_9"` with `{}` which is the delimiter used by JCasC.

Perhaps you could send a PR that adds delimiter customization, should be possible to do from the [SelfConfigurator](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/SelfConfigurator.java) and modify the [SecretResolver](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/plugin/src/main/java/io/jenkins/plugins/casc/SecretSourceResolver.java).